cpp/src/io/parquet/parquet_gpu.hpp

```diff
@@ -307,8 +308,13 @@ inline size_type __device__ row_to_value_idx(size_type idx,
       col = col.child(0);
     } else {
       auto list_col = cudf::detail::lists_column_device_view(col);
-      idx           = list_col.offset_at(idx);
-      col           = list_col.child();
+      auto child    = list_col.child();
+      if (parquet_col.output_as_byte_array &&
+          (child.type().id() == type_id::INT8 || child.type().id() == type_id::UINT8)) {
```

Comment:
This code marches down the hierarchy to find the child. I can't say if we have to do it, but we need to check this here. Otherwise we end up with the wrong number of entries.